plugins/flatpak/gs-flatpak.c

```diff
@@ -2924,16 +2924,23 @@ gs_flatpak_refine_runtime_for_install (GsFlatpak *self,
 
 static void
 gs_flatpak_add_app_to_list_if_not_installed (GsApp *app, GsAppList *list,
-					     const GHashTable *hash_installed)
+					     const GHashTable *hash_installed,
+					     gboolean include_updates)
 {
 	g_autofree gchar *ref_display = gs_flatpak_app_get_ref_display (app);
-	if (g_hash_table_contains (hash_installed, ref_display)) {
-		g_debug ("%s is already installed, so skipping",
-			 gs_app_get_unique_id (app));
+	if (g_hash_table_contains (hash_installed, ref_display) &&
+	    (!include_updates || !gs_app_is_updatable (app))) {
+		if (include_updates) {
+			g_debug ("%s is already installed and up to date, so skipping",
+				 gs_app_get_unique_id (app));
+		} else {
+			g_debug ("%s is already installed, so skipping",
+				 gs_app_get_unique_id (app));
+		}
 		return;
 	}
 
-	g_debug ("%s/%s is not already installed, so installing",
+	g_debug ("%s/%s is not already installed or needs update, so installing",
```

Comment:
I think we should either divide this message, or change it further so it doesn't say `installing`.
Perhaps:
`is not already installed/up-to-date, so adding to the list to be installed/updated`

Yeah, maybe it's better to split it.